### DISCUSSION TITLE
Returning an error object in a resolver without mercurius converting it to GraphQLError?

### DISCUSSION THREAD

**Discussion by zestsystem** [mercurius-js/mercurius#944] [Q&A]
Hi all, I want to return some errors inside "data" response of my resolvers.  However whenever I return a javascript error in a resolver, it gets converted to GraphQLError and the error that should be in userErrors array gets populated in the errors field.  I've tried wrapping the Error object following the small "workaround" listed here, but found no success with this: 
https://the-guild.dev/blog/graphql-error-handling-with-fp 

Any idea on how I should go about getting the mercurius server to ignore the error object being returned? Or more specifically, an array of custom error object extended from Error? Thank you.

**Comment by mcollina** [comment]
I'm not sure this is even doable, I don't think it's Mercurius doing the wrapping in GrapHQLError.

**Reply by zestsystem** [reply]
Thanks for the fast response! In the article linked the person uses a "small workaround" to prevent the default behavior of graphql-js like this. Which made me think that it is possible to hide and obfuscate the error object from whatever is detecting javascript error object being returned.

"A small workaround is needed because graphql-js treats error-returning and error-throwing the same in resolvers, therefore we need to wrap our Errors before we can safely return them from the resolvers. Let's use a simple custom wrapper type and constructor function: 

 '''typescript
export type WrappedError<E extends Error> = {
 _tag: 'WrappedError'
 err: NonNullable<E>
}

export const wrappedError = <E extends Error>(err: E) => ({
 _tag: 'WrappedError',
 err
})
'''

One question I have is how does the pre-execution hook get the errors from the result before adding it to the context? Also maybe something has changed in graphql-js between the time of the writing of this article and now?

**Reply by mcollina** [reply]
I don't know to be honest, I never had to debug this :(.
